src/core/middleware/store.ts

```diff
@@ -0,0 +1,54 @@
+import { destroy, invalidator, create } from '../vdom';
+import injector from '../middleware/injector';
+import Store, { StatePaths, Path } from '../../stores/Store';
+import { Process } from '../../stores/process';
+
+const factory = create({ destroy, invalidator, injector });
+
+export const createStoreMiddleware = <S = any>(initial?: (store: Store<S>) => void) => {
+	let store = new Store<S>();
+	let initialized = false;
+	initial && initial(store);
+	const storeMiddleware = factory(({ middleware: { destroy, invalidator, injector } }) => {
+		const handles: any[] = [];
+		destroy(() => {
+			let handle: any;
+			while ((handle = handles.pop())) {
+				handle();
+			}
+		});
+		if (!initialized) {
+			const injectedStore = injector.get<Store<S>>('state');
+			if (injectedStore) {
+				store = injectedStore;
+			}
+			initialized = true;
+		}
+		const registeredPaths: string[] = [];
+		const path: StatePaths<S> = (path: any, ...segments: any) => {
+			return (store as any).path(path, ...segments);
+		};
+		return {
+			get<U = any>(path: Path<S, U>, subscribe = true): U {
```

Comment:
Equivalent to the store's `get` API expect it will automatically subscribe the widget to data updates for all the paths used. This can be manually disabled on a call by call basis by passing `false` as the second argument